### PR TITLE
Inject function to set amount dynamically, Refactor middleware and LNURL

### PR DESCRIPTION
**Pull request by DhananjayPurohit**
Fix issue #12

### CODE REVIEW

File: ginlsat/ginlsat.go [hunk 1]
```diff
@@ -43,19 +43,24 @@ type LsatInfo struct {
 }
 
 type GinLsatMiddleware struct {
-	Amount   int64
-	Response func(c *gin.Context, code int, message string)
-	LNClient ln.LNClient
+	RequestForAmount *http.Request
```
Comment (kiwiidb) [inline ginlsat/ginlsat.go:46]: This should be dropped, see other comment.

**Review by kiwiidb — CHANGES_REQUESTED** [review]
Use c.Request to have a truly dynamic way of setting the amount.

File: examples/main.go [hunk 2]
```diff
@@ -35,21 +73,16 @@ func main() {
...
-	})
+	lsatmiddleware, err := ginlsat.NewLsatMiddleware(lnClientConfig, AmountFunc)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	router.Use(lsatmiddleware.Handler)
 
 	router.GET("/protected", func(c *gin.Context) {
+		c.Request = FiatToBTC("USD", 0.01)
```
Comment (kiwiidb) [inline examples/main.go:85]: You can drop this as well. It doesn't do anything, right?

Comment (DhananjayPurohit) [reply examples/main.go:85]: I wrote the function `FiatToBTC` that is providing a request to `AmountFunc` to demonstrate how the amount can be changed dynamically. So, it's creating a request for converting 0.01 USD to Sats by hitting `https://blockchain.info/tobtc?currency=USD&value=0.01` that is executing within `AmountFunc` in this example.

Comment (kiwiidb) [reply examples/main.go]: What do you think about this: https://github.com/getAlby/gin-lsat/pull/13/commits/f5d45952c4f05a0fcc06c762717ac7addac5df0c , this looks more clean to me. You shouldn't reuse the same request from c.Request just to make a request to fetch the price.

**Comment by kiwiidb** [comment]
I think tests are failing right now because your repo doesn't have the secrets.

**Review by kiwiidb — APPROVED** [review]

**Pull request merged**